api/client/commands.go

```diff
@@ -2297,6 +2297,14 @@ func (cli *DockerCli) CmdRun(args ...string) error {
 		}
 	}()
 
+	if *flAutoRemove {
+		defer func() {
+			if _, _, err := readBody(cli.call("DELETE", "/containers/"+runResult.Get("Id")+"?v=1", nil, false)); err != nil {
```

Comment:
You probably want to remove delete code below. Because it can be a problem. Also I'm not sure about just logging - we can modify returned error in defer, but I'm not sure if it's worth it.